app/redux/ducks/classify.spec.js

```diff
@@ -454,4 +456,96 @@ describe('Classifier actions', function () {
       expect(newState.classification.gold_standard).to.be.undefined;
     });
   });
+
+  describe('load workflow', function () {
+    let awaitWorkflow;
+    let state = {
+      classification: null,
+      goldStandardMode: false,
+      intervention: null,
+      upcomingSubjects: [],
+      workflow: { id: 'b' }
+    };
+    const fakeDispatch = sinon.stub().callsFake(function (action) {
+      if(typeof action === 'function') {
+        action = action(fakeDispatch);
+      }
+      state = reducer(state, action);
+      return action;
+    });
+    const fakePreferences = {
+      update: sinon.stub().callsFake(changes => changes),
+      save: sinon.stub()
+    };
+    const fakeWorkflow = {
+      id: 'a'
+    };
+
+    before(function () {
+      sinon.stub(apiClient, 'type').callsFake(function (type) {
+        if (type === 'workflows') {
+          return {
+            get: sinon.stub().callsFake(() => Promise.resolve(fakeWorkflow))
+          };
+        }
+        if (type === 'translations') {
+          const fakeTranslation = {
+            strings: {}
+          };
+          return {
+            get: sinon.stub().callsFake(() => Promise.resolve([fakeTranslation]))
+          };
+        }
+      });
+    });
+
+    after(function () {
+      apiClient.type.restore();
+    });
+
+    beforeEach(function () {
+      awaitWorkflow = loadWorkflow('a', 'en', fakePreferences)(fakeDispatch);
+    });
+
+    afterEach(function () {
+      fakeDispatch.resetHistory();
+      fakePreferences.update.resetHistory();
+      fakePreferences.save.resetHistory();
+    });
+
+    it('should clear the existing workflow', function () {
+      expect(state.workflow).to.be.null;
+    });
+
+    it('should update user preferences', function () {
+      expect(fakePreferences.update).to.have.been.calledWith({ 'preferences.selected_workflow': 'a' });
+    });
+
+    it('should load the workflow translation', function () {
+      const loadTranslations = fakeDispatch.returnValues[3];
```

Comment:
Is there a better way of spying on this call than looking for the fourth call to `dispatch()`? Sinon won't spy on pure functions, so `translations.load` can't be stubbed.